src/functions.ts

```diff
@@ -0,0 +1,112 @@
+import { FINAL_COMMA, FINAL_PAREN, PUNCTUATION } from 'consts'
+import { titleCase, swapCase, upperCaseFirst, capitalCase } from 'text-case'
+
+/**
+ * Functions get passed a mutable html anchor element and the current text
+ * They must return a new text that can be propagated to the next function
+ */
+export interface PipeFunction {
+  (link: HTMLAnchorElement, text: string): string
+}
+
+export const DEFAULT_FUNCTIONS: Array<PipeFunction> = [removeParenthesisComma, blendSuffix]
+
+export const FUNCTION_LOOKUP: Record<string, PipeFunction> = {
+  lower: lower,
+  _: lower,
+
+  upperFirst: upperFirst,
+  upper_first: upperFirst,
+  upperfirst: upperFirst,
+  '^': upperFirst,
```

Comment:
I think we should use `^` for `upper` to make it symmetric with `_` for lower.

Reply:
The reasoning behind `^` for is that the current approach follows the logic: more `^` ⇒ more upper case letters. I think this is more intuitive than symmetry with the `lower` function.

I am open to suggestions though, but I think short aliases are important - after all, pipe tricks should reduce verbosity.

One could argue that we could make one character aliases for all the important cases (and I do think sentence, title, and capital cases are all relevant). This is problematic because they will be hard to remember.

Currently the logic behind the naming of the aliases summarizes as:

- `_` - underscore is makes visually sense as the lower function
- `^` - this makes sense as turning something into a more capital case, but there are multiple of these (sentence `^` → capital `^^` → upper `^^^`)
- `#` - along with lower this will probably be the most useful function. `#` reminds me of a heading, which are often in title case.